lib/appRelationsDiscovery.js

```diff
@@ -48,20 +48,20 @@ async function getRelations(asyncApiDocs, { syntax } = defaultOptions) {
               application = relation.get(channelName);
             } else {
               application = {
-                sub: [],
-                pub: [],
+                sub: new Map(),
+                pub: new Map(),
               };
               relation.set(channelName, application);
             }
     
             const channel = doc.channel(channelName);
             const title = doc.info().title();
-    
-            if (channel.hasPublish()) {
-              application.pub.push(title);
+            
+            if (channel.hasPublish() && !application.pub.has(title)) {
+              application.pub.set(title,channel._json);
```

Comment:
Duplicated items in sub/pub will be only in case, when someone call two specs with this same title. Statements `application.pub.has(title)` should be false in 99% cases... Hmmm 🤔 Maybe we should throw the error if you have existing item, or use `id` of spec to identity app rather than `title` -> https://github.com/asyncapi/spec/blob/master/spec/asyncapi.md#A2SIdString 

Also `channel._json` is good, but better will be pass normal model from parser-js:

```suggestion
            if (channel.hasPublish() && !application.pub.has(title)) {
              application.pub.set(title,channel);
```

Then you and user can use metadata with corresponding model's functions or make `channel.json()` to retrieve model's json. It's better :)

Reply:
Yeah! that 1% could be when we would have the case we discussed earlier:
https://github.com/asyncapi/app-relations-discovery/pull/3#discussion_r647196669

> Hmmm 🤔 Maybe we should throw the error if you have existing item, or use `id` of spec to identity app rather than `title` ->
But after having the case I just linked if there's a possibility of it then we should use your approach. After all why to have these corner cases left untouched :)

Reply:
> Yeah! that 1% could be when we would have the case we discussed earlier: #3 (comment)

I think that's the different case, because in `#3 (comment)` you had problem with serverName, but there is a problem with spec/app name.

So let's throw error at the moment :)

Reply:
Ahh.. and again I thought about `serverName` 😅 You're right about:

> Yeah! that 1% could be when we would have the case we discussed earlier: #3 (comment)

But lets throw error anyway :)